framework/src/parser/Parser.C

```diff
@@ -1252,34 +1256,40 @@ Parser::setScalarParameter(const std::string & full_name,
     {
       try
       {
-        auto v = std::stoi(_root->param<std::string>(full_name));
+        auto v = std::stoi(strval, &pos);
         param->set() = *reinterpret_cast<T *>(&v);
+        if (pos != strval.size())
+          throw "foo"; // catch below and print error
       }
       catch (...)
       {
-        mooseError(errormsg(_input_filename,
+        _errmsg += errormsg(_input_filename,
                             _root->find(full_name),
                             "invalid integer syntax for parameter: ",
                             full_name,
                             "=",
-                            _root->param<std::string>(full_name)));
+                            strval) +
+                   "\n";
       }
     }
     else if (t == typeid(double))
     {
       try
       {
-        auto v = std::stod(_root->param<std::string>(full_name));
+        auto v = std::stod(strval, &pos);
         param->set() = *reinterpret_cast<T *>(&v);
+        if (pos != strval.size())
+          throw "foo"; // catch below and print error
```

Comment:
Did you really intend to slip "foo" into the production code? Also, I'd really prefer that you catch specific exceptions. I really don't like seeing global exception catchers as the first line of defense when you actually know the specific types that are thrown in the try block. This is worth cleaning up. I can see that some of this has already slipped through, will you revisit and clean it up?

Reply:
The ``...`` was because there were some exceptions that weren't being caught properly despite being std::exception subclasses.  I can change this from ``foo`` but it doesn't matter what it is because it is caught (and not rethrown) on the next line in order to avoid duplicating the error message construction/emission.

Reply:
I wonder if this problem is isolated to your machine. If you catch the right type and push it up for CIVET does it work? We really need to figure out what is going on with your system. There's no better time to go to High Sierra! 😉

Reply:
It is harder than that to deal with - all the errors that triggered that issue were removed from input files.  I can't remember what syntax triggered the missed catch (nor what exception was slipping through).  I agree it's not the best, but it seems a bit tangential to this PR (dealing with the ``...``).  Would you prefer I change the throw to something like ``std::runtime_error("dummy")``?

Reply:
Yes, I'd prefer something standard looking, but why not go even further and illustrate your intent rather than use dummy? What does it mean to have a mismatch like this? Use that definition as your string?

We can deal with `...` in another PR.

Reply:
It really does not need a different message than the one given below.  They both mean invalid number syntax.  The atoi conversion sometimes throws an exception if the syntax is super invalid or just parses only part of the string into an int if the syntax stops being an int after some point.  i.e. ``asdf`` triggers atoi to throw and ``1.3`` would trigger the manual throw.  Both are invalid integer syntax and the current message says that in addition to printing the actual value that was invalid for both cases.  I could also alternatively use ``goto`` :-)